syft/pkg/cataloger/cpe_specificity.go

```diff
@@ -1,31 +1,46 @@
 package cataloger
 
-import "github.com/facebookincubator/nvdtools/wfn"
+import (
+	"sort"
+
+	"github.com/facebookincubator/nvdtools/wfn"
+)
+
+var _ sort.Interface = (*ByCPESpecificity)(nil)
 
 type ByCPESpecificity []wfn.Attributes
 
-// Implementing sort.Interface
 func (c ByCPESpecificity) Len() int      { return len(c) }
 func (c ByCPESpecificity) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
 func (c ByCPESpecificity) Less(i, j int) bool {
-	return countSpecifiedFields(c[i]) > countSpecifiedFields(c[j])
+	iScore := countSpecifiedFields(c[i])
+	jScore := countSpecifiedFields(c[j])
+	if iScore == jScore {
+		return countFieldLength(c[i]) > countFieldLength(c[j])
+	}
+	return iScore > jScore
+}
+
+func countFieldLength(cpe wfn.Attributes) int {
+	return len(cpe.Part + cpe.Vendor + cpe.Product + cpe.Version + cpe.TargetSW)
 }
 
 func countSpecifiedFields(cpe wfn.Attributes) int {
-	checksForSpecifiedField := []func(cpe wfn.Attributes) bool{
-		func(cpe wfn.Attributes) bool { return cpe.Part != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Product != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Vendor != "" },
-		func(cpe wfn.Attributes) bool { return cpe.Version != "" },
-		func(cpe wfn.Attributes) bool { return cpe.TargetSW != "" },
+	checksForSpecifiedField := []func(cpe wfn.Attributes) (bool, int){
```

Comment:
Given this change in the function's behavior, I'd suggest updating the function name to reflect the implementation (i.e. indicate that there's more happening than just a counting of fields that are specified). I'm not opinionated on the new name but can provide ideas if that's useful